internal/iam/query.go

```diff
@@ -206,6 +206,109 @@ const (
+    ),
+    user_oidc_managed_groups (id) as (
+      select managed_group_id
+        from auth_oidc_managed_group_member_account
+       where member_id in (select id
+                             from user_accounts)
+    ),
+    user_ldap_managed_groups (id) as (
+      select managed_group_id
+        from auth_ldap_managed_group_member_account
+       where member_id in (select id
+                             from user_accounts)
+    ),
+    managed_group_roles (role_id) as (
+      select distinct role_id
+        from iam_managed_group_role
+       where principal_id in (select id
+                                from user_oidc_managed_groups)
+          or principal_id in (select id
+                                from user_ldap_managed_groups)
+    ),
+    group_roles (role_id) as (
+      select role_id
+        from iam_group_role
+       where principal_id in (select id
+                                from user_groups)
+    ),
+    user_roles (role_id) as (
+      select role_id
+        from iam_user_role
+       where principal_id in (select id
+                                from users)
+    ),
+    user_group_roles (role_id) as (
+      select role_id
+        from group_roles
+       union
+      select role_id
+        from user_roles
+       union
+      select role_id
+        from managed_group_roles
+    ),
+    roles_with_grants (role_id, canonical_grant) as (
+      select iam_role_grant.role_id,
+             iam_role_grant.canonical_grant
+        from iam_role_grant
+        join iam_role
+          on iam_role.public_id = iam_role_grant.role_id
+        join iam_grant
+          on iam_grant.canonical_grant = iam_role_grant.canonical_grant
+       where iam_role.public_id in (select role_id
+                                      from user_group_roles)
+         and iam_grant.resource = any(@resources)
+    ),
+    global_roles as (
+      select iam_role_global.public_id             as role_id,
+             iam_role_global.scope_id              as role_scope_id,
+             iam_scope.type                        as role_type,
+             'global'                              as role_parent_scope_id, -- manually set to global because we are only looking at global roles and the parent scope is always global
+             iam_role_global.grant_scope           as grant_scope,
+             iam_role_global.grant_this_role_scope as grant_this_role_scope,
+             roles_with_grants.canonical_grant     as canonical_grant
+        from iam_role_global
+        join roles_with_grants
+          on roles_with_grants.role_id = iam_role_global.public_id
+        join iam_scope
+          on iam_scope.public_id = iam_role_global.scope_id
+    )
+    select role_id,
+           role_scope_id,
+           role_parent_scope_id,
+           grant_scope,
+           grant_this_role_scope,
+           null as individual_grant_scopes, -- there can be no individual grants for global roles
```

Comment:
I think the comment here isn't quite correct. It is not that global role can't have individual grants, but rather that any individual grant scopes will not be applicable for a resource that is in the global scope.

Reply:
Updated to: `individual grant scopes do not apply to resources in the global scope`